src/applications/disability-benefits/all-claims/utils/on-submit/purge-toxic-exposure-data.js

```diff
@@ -0,0 +1,238 @@
+    if (hasNoRetainedDetails) {
+      delete result[detailsKey];
+    } else {
+      result[detailsKey] = retainedDetails;
+    }
+  }
+
+  // Remove other/specify fields when:
+  // - Field is null (orphaned data)
+  // - User selected 'none' or has no selections
+  const hasOtherField = otherKey && otherKey in result;
+  if (hasOtherField) {
+    const isOtherFieldNull = result[otherKey] === null;
+    const hasNoneSelected = result[exposureType]?.none;
+    const hasNoSelections = !hasSelectedConditions(result[exposureType]);
+
+    const shouldRemoveOtherField =
+      isOtherFieldNull || hasNoneSelected || hasNoSelections;
+    if (shouldRemoveOtherField) {
+      delete result[otherKey];
+    }
+  }
+
+  return result;
+};
+
+/**
+ * Removes orphaned and non-applicable toxic exposure data from form payload
+ * to prevent backend validation errors when users opt out of sections.
+ *
+ * This function handles scenarios where users:
+ * - Start filling sections then deselect them (leaving orphaned details)
+ * - Select "none" for conditions (complete opt-out)
+ * - Have no conditions selected (empty/missing/all false - removes all exposure data)
+ * - Remove all selections but leave partial data behind
+ * - Have null exposure fields that need cleanup
+ *
+ * Key behaviors:
+ * - Preserves unknown/unrecognized fields (forward compatibility)
+ * - Only removes data matching known orphaned patterns
+ * - Uses lodash's cloneDeep to preserve null values correctly
+ * - Processes only exposure types defined in EXPOSURE_TYPE_MAPPING
+ *
+ * NOT a validation function - this is purely for cleaning up data that
+ * shouldn't be submitted based on user's opt-out choices.
+ *
+ * @param {Object} formData - Form data to transform
+ * @param {boolean} [formData.disability526ToxicExposureOptOutDataPurge] - Feature flag for opt-out data purging
+ * @param {Object} [formData.toxicExposure] - Toxic exposure data
+ * @param {Object.<string, boolean>} [formData.toxicExposure.conditions] - Health condition selections
+ * @param {Object.<string, boolean>} [formData.toxicExposure.gulfWar1990] - Gulf War 1990 location selections
+ * @param {Object.<string, boolean>} [formData.toxicExposure.gulfWar2001] - Gulf War 2001 location selections
+ * @param {Object.<string, boolean>} [formData.toxicExposure.herbicide] - Herbicide exposure location selections
+ * @param {Object.<string, boolean>} [formData.toxicExposure.otherExposures] - Other exposure type selections
+ * @returns {Object} Form data with orphaned/non-applicable toxic exposure data removed
+ */
+export const purgeToxicExposureData = formData => {
+  if (formData.disability526ToxicExposureOptOutDataPurge !== true) {
+    return formData;
+  }
+
+  if (!formData?.toxicExposure) {
+    return formData;
+  }
+
+  const clonedData = cloneDeep(formData);
+  let { toxicExposure } = clonedData;
+
+  // Handle invalid toxicExposure types (string, array, etc.)
```

Comment:
Confused by this one, if the data is in an improper format we just return it? Will that still cause validation issues?